changelog/8509.improvement.rst

```diff
@@ -0,0 +1,3 @@
+Fixed issue where `TestCase.setUpClass` is not called when a test method has `/` in its name in versions of `pytest >=6.2.0`.
+
+Paths only consider `/` parts of a patch until the first `::` in `iterparentnodeids`. Then `::` parts of a path are considered. Test cases for this were updated to reflect this change.
```

Comment:
The changelog is user-facing, so this suggestion tries to make it more understandable for an ordinary user.

```suggestion
Fixed issue where `TestCase.setUpClass` is not called when a test has `/` in its name since pytest 6.2.0.

This refers to the path part in pytest node IDs, e.g. `TestClass::test_it` in the node ID `tests/test_file.py::TestClass::test_it`.

Now, instead of assuming that the test name does not contain ``/``, it is assumed that test path does not contain ``::``. We plan to hopefully make both of these work in the future.
```